packages/jellyfish-api-core/__tests__/category/oracle/listLatestRawPrices.test.ts

```diff
@@ -0,0 +1,255 @@
+          state: OracleRawPriceState.LIVE
+        }
+      ]
+    )
+
+    const result2 = data.filter(element => element.oracleid === oracleid2)
+
+    expect(result2).toStrictEqual(
+      [
+        {
+          priceFeeds: priceFeed2[0],
+          oracleid: oracleid2,
+          weightage: new BigNumber(2),
+          timestamp: new BigNumber(timestamp2),
+          rawprice: new BigNumber(1.5),
+          state: OracleRawPriceState.LIVE
+        },
+        {
+          priceFeeds: priceFeed2[1],
+          oracleid: oracleid2,
+          weightage: new BigNumber(2),
+          timestamp: new BigNumber(timestamp2),
+          rawprice: new BigNumber(2),
+          state: OracleRawPriceState.LIVE
+        }
+      ]
+    )
+
+    await container.call('removeoracle', [oracleid1])
+    await container.call('removeoracle', [oracleid2])
+
+    await container.generate(1)
+  })
+
+  it('should listLatestRawPrices with various timestamps', async () => {
+    const priceFeed = { token: 'APPLE', currency: 'EUR' }
+
+    const oracleid1 = await container.call('appointoracle', [await container.getNewAddress(), [priceFeed], 1])
+    const oracleid2 = await container.call('appointoracle', [await container.getNewAddress(), [priceFeed], 2])
+    const oracleid3 = await container.call('appointoracle', [await container.getNewAddress(), [priceFeed], 3])
+    const oracleid4 = await container.call('appointoracle', [await container.getNewAddress(), [priceFeed], 4])
+    const oracleid5 = await container.call('appointoracle', [await container.getNewAddress(), [priceFeed], 5])
+
+    await container.generate(1)
+
+    // NOTE(jingyi2811): Set to current time.
+    const timestamp1 = Math.floor(new Date().getTime() / 1000)
+    const prices1 = [{ tokenAmount: '0.5@APPLE', currency: 'EUR' }]
+    await container.call('setoracledata', [oracleid1, timestamp1, prices1])
+
+    // NOTE(jingyi2811): Set to 1 second before 1 hour later.
+    const timestamp2 = Math.floor(new Date().getTime() / 1000) + 3599
+    const prices2 = [{ tokenAmount: '1.0@APPLE', currency: 'EUR' }]
+    await container.call('setoracledata', [oracleid2, timestamp2, prices2])
+
+    // NOTE(jingyi2811): Set to 1 hour later.
+    const timestamp3 = Math.floor(new Date().getTime() / 1000) + 3600
+    const prices3 = [{ tokenAmount: '1.5@APPLE', currency: 'EUR' }]
+    await container.call('setoracledata', [oracleid3, timestamp3, prices3])
+
+    // NOTE(jingyi2811): Set to 1 second after 1 hour before.
+    const timestamp4 = Math.floor(new Date().getTime() / 1000) - 3599
+    const prices4 = [{ tokenAmount: '2.0@APPLE', currency: 'EUR' }]
+    await container.call('setoracledata', [oracleid4, timestamp4, prices4])
+
+    // NOTE(jingyi2811): Set to 1 hour before.
+    const timestamp5 = Math.floor(new Date().getTime() / 1000) - 3600
+    const prices5 = [{ tokenAmount: '2.5@APPLE', currency: 'EUR' }]
+    await container.call('setoracledata', [oracleid5, timestamp5, prices5])
```

Comment:
Interesting, rpc call actually can set the price for such an "old" timestamp? @monstrobishi